src/postprocessor/compress.rs

```diff
@@ -18,6 +18,44 @@ use simd_json::ValueAccess;
 use std::{io::Write, str};
 use tremor_script::Value;
 
+enum Algorithm {
+    Gzip,
+    Zlib,
+    Xz2,
+    Zstd,
+    Snappy,
+    Lz4,
+}
+
+impl Algorithm {
+    pub fn postprocessor(config: Option<&Value>, algorithm:Algorithm) -> Box<dyn Postprocessor> {
+        match config.get_str("level") {
+            Some(compression_level) => {
+                let compression_level = compression_level.parse::<i64>().unwrap(); 
+
+                let codec:Box<dyn Postprocessor> = match algorithm {
+                    Algorithm::Xz2  => Box::new(Xz2::with_config(compression_level as u32)),
+                    Algorithm::Zstd => Box::new(Zstd::with_config(compression_level as i32)),
+                    Algorithm::Lz4 => Box::new(Lz4::with_config(compression_level as u32)),
+                    _ => panic!("compression level not supported"),
```

Comment:
Lets not panic the whole process here, but make the method return a `Result<Box<dyn Postprocessor>>` and return an `Err` here.